controller/volume_controller.go

```diff
@@ -1730,6 +1729,9 @@ func (c *VolumeController) openVolumeDependentResources(v *longhorn.Volume, e *l
 		}
 		replicaAddressMap[r.Name] = imutil.GetURL(r.Status.StorageIP, r.Status.Port)
 	}
+	if !anyRunningReplica {
```

Comment:
Don't need this variable, because you can just rely on the below condition. If `len(replicaAddressMap) != 0` already means there is at least one healthy inside?

Reply:
Yap, I am also thinking about reusing this condition.
But I felt that checking is error handling instead retry. Then I choose to add a new checking variable.

The original flow is more like:
1. wait for whole replica running
2. if the whole replica running we should not have the situation that replicaAddressMap is empty. So if we meet this situation, just return the error.

If I use the `replicaAddressMap` for checking, the new flow would like
1. keep checking until at least one replica is healthy.
Then we could remove the running checks.

I thought maybe the second would be enough, but originally I would like to keep more code flow as before.
So what do you think about the second change (means only relying on `replicaAddressMap`)?